src/proc/align.cpp

```diff
@@ -13,6 +13,373 @@
+                _pre_compute_map_y_bottom_right.data(), (byte*)_pixel_bottom_right_int.data(), to, from_to_other);
+
+            bottom_right = _pixel_bottom_right_int;
+        }
+
+        switch (bpp)
+        {
+        case 1:
+            move_other_to_depth(z_pixels, reinterpret_cast<const bytes<1>*>(source), reinterpret_cast<bytes<1>*>(dest), to,
+               _pixel_top_left_int, bottom_right);
+            break;
+        case 2:
+            move_other_to_depth(z_pixels, reinterpret_cast<const bytes<2>*>(source), reinterpret_cast<bytes<2>*>(dest), to,
+                _pixel_top_left_int, bottom_right);
+            break;
+        case 3:
+            move_other_to_depth(z_pixels, reinterpret_cast<const bytes<3>*>(source), reinterpret_cast<bytes<3>*>(dest), to,
+                _pixel_top_left_int, bottom_right);
+            break;
+        default:
+            break;
+        }
+    }
+
+    template<class T >
+    void image_transform::move_other_to_depth(const uint16_t* z_pixels,
+        const T* source,
+        T* dest, const rs2_intrinsics& to,
+        const std::vector<int2>& pixel_top_left_int,
+        const std::vector<int2>& pixel_bottom_right_int)
+    {
+        // Iterate over the pixels of the depth image
+        for (int y = 0; y < _depth.height; ++y)
+        {
+            for (int x = 0; x < _depth.width; ++x)
+            {
+                auto depth_pixel_index = y*_depth.width + x;
+                // Skip over depth pixels with the value of zero, we have no depth data so we will not write anything into our aligned images
+                if (z_pixels[depth_pixel_index])
+                {
+                    for (int other_y = pixel_top_left_int[depth_pixel_index].y; other_y <= pixel_bottom_right_int[depth_pixel_index].y; ++other_y)
+                    {
+                        for (int other_x = pixel_top_left_int[depth_pixel_index].x; other_x <= pixel_bottom_right_int[depth_pixel_index].x; ++other_x)
+                        {
+                            if (other_x < 0 || other_y < 0 || other_x >= to.width || other_y >= to.height)
+                                continue;
+                            auto other_ind = other_y * to.width + other_x;
+
+                            dest[depth_pixel_index] = source[other_ind];
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    template<rs2_distortion dist>
+    inline void image_transform::distorte_x_y(const __m128 & x, const __m128 & y, __m128 * distorted_x, __m128 * distorted_y, const rs2_intrinsics& to)
+    {
+        *distorted_x = x;
+        *distorted_y = y;
+    }
+    template<>
+    inline void image_transform::distorte_x_y<RS2_DISTORTION_MODIFIED_BROWN_CONRADY>(const __m128& x, const __m128& y, __m128* distorted_x, __m128* distorted_y, const rs2_intrinsics& to)
+    {
+        __m128 c[5];
```

Comment:
Please add comments for the hard-coded values - [5] == distortion coefficients,
for maintainability